File: api/src/main/java/api/maps/UserMapper.java
package api.maps;

//java lib
import java.util.function.Function;
//spring lib
import org.springframework.stereotype.Service;
import api.entities.UserEntity;
import api.dtos.UserDTO;


@Service
public class UserMapper implements Function<UserEntity, UserDTO>{
    @Override
    public UserDTO apply(UserEntity u){
      return new UserDTO(u.getName(), u.getEmail());
    }
}

File: api/src/main/java/api/maps/UserMapper.kt
package api.maps

import api.dtos.UserDTO
import api.dtos.UserRegisterationDto
import api.entities.UserEntity

fun UserDTO.toEntity(withPassword: String): UserEntity = UserEntity(
    id = this.id,
    username = this.username,
    email = this.email,
    password = withPassword,
    firstname = firstname,
    lastname = this.lastname,
    creationDate = this.creationDate
)

fun UserEntity.toDto(): UserDTO = UserDTO(
    id = this.id,
    username = this.username,
    email = this.email,
    firstname = this.firstname,
    lastname = this.lastname,
    creationDate = this.creationDate
)

fun UserRegisterationDto.toEntity(): UserEntity = UserEntity(
    username = this.username,
    email = this.email,
    password = this.password
)
